### PR TITLE
bzlmod: Add support for WORKSPACE.bzlmod

### PR DESCRIPTION
**Pull request by meteorcloudy**
When enabling Bzlmod, if a WORKSPACE.bzlmod file is presented at the
source root, it will replace the content of the WORKSPACE file.
In contrast to the WORKSPACE file, no prefix or suffix are added to the WORKSPACE.bzlmod file.

This file is for migration purpose in case some of the dependencies
still need to be introduced via the WORKSPACE mechanism, eventually
it should be removed.

RELNOTES: None
GOOGLE:

### CODE REVIEW

File: WORKSPACE.bzlmod [hunk 1]
```diff
@@ -1,31 +1,425 @@
-# Replace the original WORKSPACE file with this file to build Bazel with bzlmod.
-# We still need the bind rules to make protobuf work.
+# Replace the original WORKSPACE file with this file to build Bazel with Bzlmod.
```
Comment (Wyverald) [inline WORKSPACE.bzlmod:1]: this line is no longer necessary, right?

File: src/main/java/com/google/devtools/build/lib/skyframe/WorkspaceFileFunction.java [hunk 2]
```diff
@@ -100,27 +101,49 @@ public SkyValue compute(SkyKey skyKey, Environment env)
...
+
+    Optional<RootedPath> resolvedFile = Preconditions.checkNotNull(
+        RepositoryDelegatorFunction.RESOLVED_FILE_INSTEAD_OF_WORKSPACE.get(env));
+    boolean useWorkspaceResolvedFile = resolvedFile.isPresent();
+
+    boolean useWorkspaceBzlmodFile = false;
+    RootedPath workspaceBzlmodFile = RootedPath.toRootedPath(workspaceFile.getRoot(),
+        workspaceFile.getRootRelativePath().replaceName("WORKSPACE.bzlmod"));
+    // We only need to check WORKSPACE.bzlmod when the resolved file isn't used.
+    if (!useWorkspaceResolvedFile) {
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/skyframe/WorkspaceFileFunction.java:121]: this condition can be simply rewritten as:

```java
if (!useWorkspaceResolvedFile && RepositoryDelegatorFunction.ENABLE_BZLMOD.get(env)) {
  FileValue ... = ...;
  if (...) { ... }
  useWorkspaceBzlmodFile = ...;
}
```

IMO the logic is much clearer that way.

File: WORKSPACE.bzlmod (hunk 1 above)
Comment (meteorcloudy) [reply WORKSPACE.bzlmod:1]: Oh, right...

File: src/main/java/com/google/devtools/build/lib/skyframe/WorkspaceFileFunction.java (hunk 2 above)
Comment (meteorcloudy) [reply src/main/java/com/google/devtools/build/lib/skyframe/WorkspaceFileFunction.java:121]: Oh, absolutely,, thanks!

File: WORKSPACE.bzlmod (hunk 1 above)
Comment (meteorcloudy) [reply WORKSPACE.bzlmod:1]: Indeed..

**Comment by meteorcloudy** [comment]
Merged in https://github.com/bazelbuild/bazel/commit/09fe7cce9a2b8716476b001be221c42d3d03b180#diff-98160be0a821b608863421665e00bc0af7276035713a2f309a19fdd426117faf

**Pull request closed without merging**